From e9f253c4ce23883f60e144c660e33ae40cf3d655 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Sun, 28 Feb 2021 18:24:45 -0800
Subject: [PATCH] Fall back to gettext module on systems that don't use glibc

Python built without gettext support (from glibc) fail here, so this
falls back to importing the gettext python implementation
(https://github.com/hannosch/python-gettext), which might
be available.
---
 src/dev.tchx84.Portfolio.in | 9 +++++++--
 src/places.py               | 5 ++++-
 src/window.py               | 6 +++++-
 3 files changed, 16 insertions(+), 4 deletions(-)

diff --git a/src/dev.tchx84.Portfolio.in b/src/dev.tchx84.Portfolio.in
index 9a9068a..6d84afd 100755
--- a/src/dev.tchx84.Portfolio.in
+++ b/src/dev.tchx84.Portfolio.in
@@ -29,8 +29,13 @@ localedir = '@localedir@'
 sys.path.insert(1, pkgdatadir)
 signal.signal(signal.SIGINT, signal.SIG_DFL)
 
-locale.bindtextdomain('portfolio', localedir)
-locale.textdomain('portfolio')
+try:
+    locale.bindtextdomain('portfolio', localedir)
+    locale.textdomain('portfolio')
+except AttributeError as e:
+    import gettext
+    gettext.bindtextdomain('portfolio', localedir)
+    gettext.textdomain('portfolio')
 
 if __name__ == '__main__':
     from gi.repository import Gio
diff --git a/src/places.py b/src/places.py
index 7d682cb..6946f94 100644
--- a/src/places.py
+++ b/src/places.py
@@ -17,7 +17,10 @@
 
 import os
 
-from locale import gettext as _
+try:
+    from locale import gettext as _
+except ImportError:
+    from gettext import gettext as _
 
 from gi.repository import GLib, Gio, Gtk, GObject
 
diff --git a/src/window.py b/src/window.py
index bd4448e..d987227 100644
--- a/src/window.py
+++ b/src/window.py
@@ -18,7 +18,11 @@
 import os
 
 from pathlib import Path
-from locale import gettext as _
+
+try:
+    from locale import gettext as _
+except ImportError:
+    from gettext import gettext as _
 
 from gi.repository import Gtk, GLib, Gio, Handy
 
-- 
2.30.1

